### PR TITLE
statistics: use MurmurPool to reuse the memory

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/47016

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by qw4990 — APPROVED** [review]

File: statistics/fmsketch.go [hunk 1]
```diff
@@ -88,31 +92,36 @@ func (s *FMSketch) InsertValue(sc *stmtctx.StatementContext, value types.Datum)
...
 		return errors.Trace(err)
 	}
-	s.hashFunc.Reset()
-	_, err = s.hashFunc.Write(bytes)
+	hashFunc := murmur3Pool.Get().(hash.Hash64)
+	hashFunc.Reset()
+	defer murmur3Pool.Put(hashFunc)
+	_, err = hashFunc.Write(bytes)
 	if err != nil {
+		murmur3Pool.Put(hashFunc)
```
Comment (time-and-fate) [inline statistics/fmsketch.go:100]: `Put()` is called twice on `hashFunc`.

Comment (hawkingrei) [reply statistics/fmsketch.go]: Fixed.

**Review by time-and-fate — APPROVED** [review]

**Pull request merged**